ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -182,19 +259,33 @@ instance SameDepIndex (BlockQuery blk) => SameDepIndex (Query blk) where
     = Just Refl
   sameDepIndex GetSystemStart _
     = Nothing
+  sameDepIndex GetChainBlockNo GetChainBlockNo
+    = Just Refl
+  sameDepIndex GetChainBlockNo _
+    = Nothing
+  sameDepIndex GetChainPoint GetChainPoint
+    = Just Refl
+  sameDepIndex GetChainPoint _
+    = Nothing
 
 deriving instance Show (BlockQuery blk result) => Show (Query blk result)
 
 -- | Answer the given query about the extended ledger state.
-answerQuery ::
-     (QueryLedger blk, ConfigSupportsNode blk)
+answerQuery :: forall blk result.
+     (QueryLedger blk, ConfigSupportsNode blk, HasAnnTip blk)
   => ExtLedgerCfg blk
   -> Query blk result
   -> ExtLedgerState blk
   -> result
 answerQuery cfg query st = case query of
   BlockQuery blockQuery -> answerBlockQuery cfg blockQuery st
   GetSystemStart -> getSystemStart (topLevelConfigBlock (getExtLedgerCfg cfg))
+  GetChainBlockNo -> case headerStateTip (headerState st) of
+    Origin -> Origin
+    At hst -> At (annTipBlockNo hst)
```

Comment:
Can we make this a helper function, e.g. `headerStateBlock`? Put it next to the existing `headerStatePoint` function.

Reply:
Done!